src/codegen/codegen_c_visitor.cpp

```diff
@@ -990,7 +984,7 @@ std::string CodegenCVisitor::get_parameter_str(const ParamVector& params) {
                                     std::get<1>(*iter),
                                     std::get<2>(*iter),
                                     std::get<3>(*iter));
-        if (!utils::is_last(iter, params)) {
+        if (!nmodl::utils::is_last(iter, params)) {
```

Comment:
There are 2 `utils` namespaces.
There is the `nmodl::utils` which includes general utility functions in `nmodl`.
And there is `nmodl::codegen::utils` that includes the utility functions only for the `codegen`.
So, in this case we have to explicitly use the full `namespace` name to disambiguate them.

Reply:
Ahh didn’t realise that! Wouldn’t it be confusing ? i.e. should we rename to codegen_utils or something ? Just asking because I don’t  know what is current best practices is. @alkino?

Reply:
The best practice is to always give full namespace path ^^

Maybe we can group functions in only one namespace?

Or not create a `is_last` and use something more "STL".

Reply:
If you want the codegen_utils, you only want to do
`using codegen_utils = namespace nmodl::codegen::utils`

Reply:
Ok